Add support for additional controllers to SaveMii's menus

jakobkg opened the pull request:
These changes allow for any wireless nintendo controllers to be used in SaveMii, adding support for the wiimote, wii classic controllers, wii pro controllers and wii u pro controllers in addition to the gamepad.

jakobkg committed:
[WIP] Add support for additional controllers

Initial commit, currently broken. Bloated lib_easy by ~1000 lines. No buttons work, the unmodified code for the sticks still does.

jakobkg committed:
Implemented a check for if any button is pressed

The parameter isPressed(PAD_BUTTON_ANY) replaces the old isPressed(0xFFFF), fixing the last issues with this branch

jakobkg committed:
Combined handling of pro and classic controllers

Turns out the two use the esact same bitmapping for everything, with the pro controller having two extra bits for R3 and L3. Merged the two for slightly shorter code.

jakobkg committed:
Restored lib_easy and moved controller things to own files

Maybe bloating lib_easy isn't a great idea, I've restored it and commented out all the controller bits, and moved my controller things to new files.

Ryuzaki-MrL reviewed (changes requested):
This can be further simplified by having a function that receives one of the arrays (buttons_pressed, buttons_hold, buttons_released) and runs the checks on it.

jakobkg committed:
Simplified the controller functions

The three functions related to updating the arrays of buttons tates are now a single function, updateButtons.

The three functions isPressed(), isHeld(), and isReleased() have been merged into a single function, checkButton() which takes parameters for which button to check and what state or event (PRESS, HOLD, RELEASE) to check for.

jakobkg commented:
There's now a single function, checkButton() that takes a button and a state as parameters, and runs checks for that button in that state (e.g. checkButton(PAD_BUTTON_B, PRESS) to see if the B button was pressed on a controller). Good idea, this cuts the code length for controllers by about 70% from my first commit. Thank you for your feedback!
